schema/namespace.go

```diff
@@ -0,0 +1,62 @@
+package schema
+
+import (
+	"strings"
+)
+
+const (
+	// thingfulSchema is the base url for Thingful's ontology
+	thingfulSchema = "https://thingful.github.io/schema#"
+
+	// m3liteSchema is the base url for the m3-lite ontology
+	m3liteSchema = "http://purl.org/iot/vocab/m3-lite#"
+
+	// ssnSchema is the base url for the SSN ontology
+	ssnSchema = "http://purl.oclc.org/NET/ssnx/ssn#"
+
+	// iotliteSchema is the base url for the IOT-LITE ontoloty
+	iotliteSchema = "http://purl.oclc.org/NET/UNIS/fiware/iot-lite#"
+
+	// xsdSchema is the base url for XSD terms
+	xsdSchema = "http://www.w3.org/2001/XMLSchema#"
+)
+
+var (
+	// expanded is a map containing the schema to compact form mappings.
+	namespaces = map[string]string{
+		"thingful:": thingfulSchema,
+		"m3-lite:":  m3liteSchema,
+		"ssn:":      ssnSchema,
+		"iot-lite:": iotliteSchema,
+		"xsd:":      xsdSchema,
+	}
+)
+
+// Expand returns the expanded version of a value (either prop or value). If we
+// are unable to expand the value then we return the input string. This allows
+// this function to be called safely on an already expanded property.
+func Expand(val string) string {
+	// k is the compact version, v is the expanded, and val is a compact string
+	for k, v := range namespaces {
+		if strings.Contains(val, k) {
+			return strings.Replace(val, k, v, 1)
+		}
+	}
+
```

Comment:
Since we know the expanded `val` should be a URL it could be worth adding a check here to ensure the returned value is a well formed URL.